src/Avalonia.Base/Data/BindingOperations.cs

```diff
@@ -33,9 +34,84 @@ public static class BindingOperations
 
             var mode = binding.Mode;
 
+            AvaloniaPropertyMetadata? metadata = null;
+
             if (mode == BindingMode.Default)
             {
-                mode = property.GetMetadata(target.GetType()).DefaultBindingMode;
+                metadata = property.GetMetadata(target.GetType());
+
+                mode = metadata.DefaultBindingMode;
+            }
+
+            var updateSourceTrigger = binding.UpdateSourceTrigger;
+
+            if (updateSourceTrigger == UpdateSourceTrigger.Default)
+            {
+                metadata ??= property.GetMetadata(target.GetType());
+
+                updateSourceTrigger = metadata.UpdateSourceTrigger;
+            }
+
+            static IObservable<object?> GetTargetPropertyObservable(
+                InstancedBinding binding,
+                IAvaloniaObject target,
+                AvaloniaProperty property,
+                UpdateSourceTrigger updateSourceTrigger)
+            {
+                var explicitSourceUpdate = binding
+                    .ExplicitUpdateRequested
+                    .Where(m => m == InstancedBinding.ExplicitUpdateMode.Source)
+                    .Select(_ => target.GetValue(property));
+
+                IObservable<object?> valueObservable;
+
+                switch (updateSourceTrigger)
+                {
+                    case UpdateSourceTrigger.PropertyChanged:
+                    case UpdateSourceTrigger.Default:
+
+                        valueObservable = target.GetObservable(property);
+
+                        break;
+
+
+                    case UpdateSourceTrigger.Explicit:
+
+                        return explicitSourceUpdate;
+
+
+                    case UpdateSourceTrigger.LostFocus:
+
+                        valueObservable = Observable.FromEventPattern<AvaloniaPropertyChangedEventArgs>(
```

Comment:
As mentioned in the comments, this can be changed to be made less hacky once the core library merge is done. Maybe leave a comment here?

```suggestion
                        // HACK: Use the LostFocus event here once core libraries are merged and
						// we have access to InputElement.
                        valueObservable = Observable.FromEventPattern<AvaloniaPropertyChangedEventArgs>(
```

Reply:
Are we planning to even merge this _before_ the core library merge happens?

Reply:
Core libraries are merged now